### DISCUSSION TITLE
Twitter like navigation layout

### DISCUSSION THREAD

**Discussion by SumitR9910** [grahammendick/navigation#823] [Q&A] [answered]
![aa204617-754f-477f-aa3d-7b4638b57253](https://github.com/user-attachments/assets/68102721-f455-415e-be0a-2b5094a36629)

Can anyone help me implement this type of navigation layout where a non-primary tab bar is nested inside of a primary tab bar item, as show above in the Twitter app  (the for you and following tabs nested inside the main home tab).

I was not able to replicate this, as nesting a non primary tab bar inside of a primary tab bar item, simply does not render it.

**Comment by grahammendick** [comment, marked as answer]
Hi @SumitR9910 nice to hear from you. I've actually [written a twitter sample](https://github.com/grahammendick/navigation/tree/master/NavigationReactNative/sample/twitter) with a [non-primary `TabBar` on the Notifications scene](https://github.com/grahammendick/navigation/blob/713feb80124cc57bb4f51ba19c95ebe1ec00db3a/NavigationReactNative/sample/twitter/Notifications.js#L19) inside a [primary `TabBar`](https://github.com/grahammendick/navigation/blob/713feb80124cc57bb4f51ba19c95ebe1ec00db3a/NavigationReactNative/sample/twitter/Tabs.js#L24).

Maybe the problem you're having is that `TabBarItems` can only have a single (native) child. If that child is a `View` you should set the `collapsable` prop to false so that you don't accidentally end up with more than one child on the native side.

**Comment by SumitR9910** [comment]
Thanks for the fast reply.

I solved this problem by rendering the non-primary tab bar inside of a stack of its own i.e a stack nested inside of the tabBar item which contains the non-primary tab bar. It is working as intended.

Although styling the tabBar has some issues, like the unselectedTintColor prop does not work (I am testing on android).
I tried changing the theme in the styles.xml to Material 3 which caused build issues in expo.
Do you have any idea why this is happening?

I am trying to migrate from Expo Router, so sorry if I seem clueless.

**Reply by grahammendick** [reply]
I'm glad you got it working.
I set the `unselectedTintColor="red"` on both primary and non-primary in the twitter sample and you can that both work in screenshot below. Maybe you can try that in the twitter sample to check that it works for you, too?

![image](https://github.com/user-attachments/assets/9d56786c-0119-43ca-a9e3-a9e3c583b623)

**Reply by grahammendick** [reply]
@SumitR9910 are you getting on ok or are you stuck with anything?
